drivers/i2c/i2c_dw.c

```diff
@@ -635,6 +637,21 @@ static int i2c_dw_transfer(const struct device *dev,
 		return ret;
 	}
 
+	/*
+	 * I2C transfer with 0 length are invalid but
+	 * SMbus quick commands uses 0 length transfer.
+	 * And in order to support SMbus quick commands, the driver
+	 * has to be enhanced.
+	 */
```

Comment:
This sounds like a feature appropriate for `drivers/smbus/`, not `drivers/i2c/`? Same goes for the Kconfig introduced?

Reply:
Few of the I2C IPs supports Quick Commands or 0 length i2c commands based on the hardware configuration.